src/test/java/io/rtr/conduit/AMQPIntegrationIT.java

```diff
@@ -108,17 +118,75 @@ private static Publisher buildPublisher() {
                 .password("guest")
                 .exchange(EXCHANGE)
                 .routingKey(ROUTING_KEY)
+                .automaticRecoveryEnabled(true)
                 .build();
     }
 
     @Test
     public void testSslAmqpTransport() {
-        callback = mock(AMQPConsumerCallback.class);
+        AMQPConsumerCallback callback = mock(AMQPConsumerCallback.class);
         AMQPMessageBundle message = new AMQPMessageBundle("a message");
         withSystemProperty("javax.net.ssl.trustStore", TRUST_STORE, () -> {
-            connectAndPublish(buildPublisher(), buildConsumer(callback), message);
+            Publisher publisher = buildPublisher();
+            Consumer consumer = buildConsumer(callback);
+            connectResources(publisher, consumer);
+            publishMessage(publisher, message);
+
+            Mockito.verify(callback, timeout(500).times(1)).handle(any());
+
+            try {
+                consumer.close();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        });
+    }
 
-            Mockito.verify(callback, times(1)).handle(any());
+    @Test
+    public void testReconnectAfterBrokerShutdown() {
+        AMQPConsumerCallback callback = mock(AMQPConsumerCallback.class);
+        AMQPMessageBundle message = new AMQPMessageBundle("a message");
+        withSystemProperty("javax.net.ssl.trustStore", TRUST_STORE, () -> {
+            Publisher publisher = buildPublisher();
+            Consumer consumer = buildConsumer(callback);
+            connectResources(publisher, consumer);
+
+            BROKER.shutdown();
+            assertFalse(publisher.isConnected());
+            assertFalse(consumer.isConnected());
+            Mockito.verify(callback).notifyOfShutdown(any(), any());
+
+            try {
+                BROKER.startup(OPTIONS);
+                assertResourcesConnect(publisher, consumer);
+            } catch (InterruptedException e) {
+                throw new AssertionError("Publisher or consumer failed to reconnect!", e);
+            }
+            catch (Exception e) {
+                throw new AssertionError("Failed to restart the broker", e);
+            }
+
+            publishMessage(publisher, message);
+            Mockito.verify(callback, timeout(2000).times(1)).handle(any());
+
+            try {
+                consumer.close();
+            } catch (IOException ignored) {}
         });
     }
+
+    private void assertResourcesConnect(Publisher publisher, Consumer consumer) throws InterruptedException {
+        final CountDownLatch done = new CountDownLatch(1);
+
+        new Thread(() -> {
+            try {
+                if ((publisher == null || publisher.isConnected()) && (consumer == null || consumer.isConnected())) {
+                    done.countDown();
+                }
+                TimeUnit.MILLISECONDS.sleep(100);
```

Comment:
did you mean to wrap this in a loop and sleep only if not yet connected? Right now the main thread will always wait 10 seconds unless the consumer and publisher connected almost immediately

Reply:
Whoops